### ISSUE TITLE
Null Reference Exception generating views using Power Tools

### ISSUE THREAD

**Issue by rowanmiller** [aspnet/EntityFramework6#15]
When using Visual Studio 2012 with Update 5 and EF 6.1.3 and the EF 6 Power Tools, when I click on 
 Entity Framework > Generate Views, I get a NullReferenceException in Microsoft.DbcontextPackage.Handlers.OptimizeContextHandler.OptimizeEmdx 

 The NullReferenceException was the result of the type resolution service not being able to resolve the EF6 assembly (ef6Assembly == null).

ByteBuilder wrote Dec 8, 2015 at 1:53 PM [x] 

The proposed fix is to OptimizeContextHandler.cs in the PowerTools project is (from line 106):

```
                var typeService = _package.GetService<DynamicTypeService>();
                var solution = _package.GetService<SVsSolution, IVsSolution>();
                IVsHierarchy hierarchy;
                solution.GetProjectOfUniqueName(project.UniqueName, out hierarchy);
                var typeResolutionService = typeService.GetTypeResolutionService(hierarchy);
                var ef6Assembly = typeResolutionService.GetAssembly(
                    new AssemblyName { Name = ef6Reference.Name, Version = new Version(ef6Reference.Version) });

                    if (ef6Assembly != null)
                    {
                        string containerName;
                        var mappingCollection = edmxUtility.GetMappingCollectionEF6(ef6Assembly, out containerName);
                        var contextTypeName = selectedItem.ProjectItem.GetDefaultNamespace() + "." + containerName;

                        OptimizeContextEF6(languageOption, baseFileName, mappingCollection, selectedItem, contextTypeName);
                    }
                    else
                    {
                        var mappingCollection = edmxUtility.GetMappingCollection();

                        OptimizeContextEF5(languageOption, baseFileName, mappingCollection, selectedItem);
                    }
```

RoMiller wrote Feb 22 at 11:42 AM [x] 

Note for Triage: In our last triage we couldn't remember if we had APIs in EF6 for this - we do https://msdn.microsoft.com/en-us/data/dn469601#code.

**Comment by rowanmiller** [comment]
Ported from CodePlex https://entityframework.codeplex.com/workitem/2862

**Comment by lajones** [comment]
Fix checked into master with commit 9d7ef534.
